src/librustc/ty/mod.rs

```diff
@@ -558,9 +558,21 @@ impl<'tcx> serialize::UseSpecializedDecodable for Ty<'tcx> {}
 /// the same contents can exist in the same context.
 /// This means we can use pointer + length for both
 /// equality comparisons and hashing.
-#[derive(Debug, RustcEncodable)]
+#[derive(RustcEncodable)]
 pub struct Slice<T>([T]);
 
+impl<T: fmt::Debug> fmt::Debug for Slice<T> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        // fmt::Display::fmt(&self.name, fmt)
+        let slice: &[T] = &*self;
+        write!(fmt,
+               "{:?} (addr={:x}, ptr={})",
+               slice,
+               self.as_ptr() as usize,
+               self.len())
```

Comment:
If you want to keep this, it should check the `verbose` flag via `ty::tls` - also, you meant `len` instead of `ptr`, right?

Reply:
hmm. that would be verbose indeed. I'd almost rather *yet another* flag -- but really what we need to do is make `{:?}` and (today's) `-Zverbose` equivalent. That is, I had hoped that `-Zverbose` would just make `{}` and `{:?}` do the same thing, not go altering debug to dump *yet more* information.

Reply:
Whoops, that was not supposed to be part of the comment. It's a leftover from debugging. Unless you thing it is generally useful, I'd just remove it.